pilot/adapter/config/file/monitor.go

```diff
@@ -103,22 +102,19 @@ func NewMonitor(delegateStore model.ConfigStore, rootDirectory string, types []s
 	return monitor
 }
 
-// Start launches a thread that monitors files under the root directory, updating the underlying config
-// store when changes are detected.
-func (m *Monitor) Start(ctx context.Context) {
+// Run monitors files under the root directory, updating the underlying config store when changes are detected.
+func (m *Monitor) Run(stop chan struct{}) {
 	m.checkAndUpdate()
 	tick := time.NewTicker(m.checkDuration)
-	go func() {
-		for {
-			select {
-			case <-ctx.Done():
-				tick.Stop()
-				return
-			case <-tick.C:
-				m.checkAndUpdate()
-			}
+	for {
+		select {
+		case <-stop:
+			tick.Stop()
+			return
+		case <-tick.C:
+			m.checkAndUpdate()
 		}
-	}()
+	}
```

Comment:
Can you add a readme in this folder explaining how people can use the file based adapter without the api server?

Reply:
done.